base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -403,14 +403,27 @@ where T: BlockchainBackend + 'static
             NodeCommsRequest::FetchMmrNodes(tree, pos, count, hist_height) => {
                 let mut added = Vec::<Vec<u8>>::with_capacity(*count as usize);
                 let mut deleted = Bitmap::create();
-                let mmr_nodes =
-                    async_db::fetch_mmr_nodes(self.blockchain_db.clone(), *tree, *pos, *count, Some(*hist_height))
-                        .await?;
-                for (index, (leaf_hash, deletion_status)) in mmr_nodes.into_iter().enumerate() {
-                    added.push(leaf_hash);
-                    if deletion_status {
-                        deleted.add(*pos + index as u32);
-                    }
+                match async_db::fetch_mmr_nodes(self.blockchain_db.clone(), *tree, *pos, *count, Some(*hist_height))
+                    .await
+                {
+                    Ok(mmr_nodes) => {
+                        for (index, (leaf_hash, deletion_status)) in mmr_nodes.into_iter().enumerate() {
+                            added.push(leaf_hash);
+                            if deletion_status {
+                                deleted.add(*pos + index as u32);
+                            }
+                        }
+                        deleted.run_optimize();
+                    },
+                    // We need to suppress the error as another node might ask for mmr nodes we dont have, so we
+                    // return ok([])
+                    Err(e) => warn!(
+                        target: LOG_TARGET,
+                        "Could not provide requested mmr nodes (pos:{},count:{}) to peer because: {}",
+                        pos,
+                        count,
+                        e.to_string()
+                    ),
                 }
                 deleted.run_optimize();
```

Comment:
Looks like this should also be reverted?

Reply:
I think that was common to both versions